src/Commands/MakeEnum.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Spatie\Enum\Laravel\Commands;
+
+use Illuminate\Console\GeneratorCommand;
+use Illuminate\Support\Str;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+
+class MakeEnum extends GeneratorCommand
+{
+    protected $name = 'make:enum';
+    protected $description = 'Create a new enum';
+
+    protected function getStub()
+    {
+        return __DIR__.'/../../stubs/enum.stub';
+    }
+
+    protected function getDefaultNamespace($rootNamespace)
+    {
+        return $rootNamespace.'\Enums';
+    }
+
+    protected function replaceClass($stub, $name)
+    {
+        $stub = parent::replaceClass($stub, $name);
+
+        $stub = str_replace('DummyDocBlock', $this->getDocBlock(), $stub);
+        $stub = str_replace('DummyValueMapConst', $this->getValueMapConst(), $stub);
+
+        return $stub;
+    }
+
+    protected function getDocBlock(): string
+    {
+        $methods = array_merge($this->option('method'), $this->option('value'));
+
+        if(!empty($methods)) {
+            $docBlock = PHP_EOL.'/**';
+            $docBlock .= implode('', array_map(function($method) {
+                    return PHP_EOL.' * @method static self '.$this->formatValueToMethod($method).'()';
+                }, $methods));
+            $docBlock .= PHP_EOL.' */';
+        }
+
+        return $docBlock ?? '';
+    }
+
+    protected function getValueMapConst(): string
+    {
+        $values = $this->option('value');
+
+        if(!empty($values)) {
+            $tab = str_repeat(' ', 4);
+            $constant = $tab.'const MAP_VALUE = [';
+
+            foreach ($values as $value) {
+                $constant .= PHP_EOL.$tab.$tab.'\''.$this->formatValueToMethod($value).'\' => \''.$value.'\',';
+            }
+
+            $constant .= PHP_EOL.$tab.'];';
+        }
+
+        return $constant ?? '';
+    }
+
+    protected function formatValueToMethod(string $value): string
+    {
+        $formatter = $this->getFormatter();
+
+        return $this->option('formatter') === 'const' ? strtoupper(Str::$formatter($value)) : Str::$formatter($value);
+    }
+
+    protected function getFormatter(): string
+    {
+        $formatter = $this->option('formatter');
+
+        if ($formatter === 'const') {
```

Comment:
It's a little confusing that `const` get converted to `snake` here, yet `formatValueToMethod` checks for it again. 

Can you think of other ways to make this more clear? I understand why it's done, but maybe there's a clearer approach.

Reply:
The only solution I could imagine would be a switch-case. This would also be more readable I think.

Reply:
Let's go with the switch then :)

Reply:
https://github.com/spatie/laravel-enum/pull/12/commits/e94163dc948d68985ac4e6e19f58e7294693fc9f